Renaming no extension "File"s containing plaintext to .MD webview arrow

<!-- NOTE! The information you specify will be publicly accessible. -->
### Expected behaviour
-making a "File" and naming it any version of "readme.md" then trying to open (which defaults to web-view)

### Actual behaviour
- opening a markdown file made from renaming a "File" containing some text within Defold produces an error message
- right-click > Open As > Text still works correctly

### Steps to reproduce

I'm making a "collection of brain teaser games" game (think Lumosity, Peak, Elevate specifically, or Warioware / Mario Party for real examples of collections of games.

- Make a folder for each game and associated bits and pieces, e.g.
BrainTrain
>DirectionLook
>>DirectionLook.collection, 
>>Arrow.go
>>Arrow.script
>>readme
- inside each minigame's folder, create a New File or New > File, name it readme
- put some text into it
- rename it readme.md
- try and open
- unexpected result

<hr/>

<table>
<tr><td>Build time</td><td>2024-02-15T08:54:48.112358</td></tr>
<tr><td>Defold channel</td><td>editor-alpha</td></tr>
<tr><td>Defold editor sha</td><td>7bafadda047312e68a7094c2928e5a403ebf90d8</td></tr>
<tr><td>Defold engine sha</td><td>fcec9500fd438e365d4e4f6cec24b99afb907247</td></tr>
<tr><td>Defold version</td><td>1.6.4</td></tr>
<tr><td>Error</td><td><a href='https://sentry.io/organizations/defold/issues/?query=id%3A"b0c6a27aacee4579adb901e5b464e850"'>b0c6a27aacee4579adb901e5b464e850</a></td></tr>
<tr><td>GPU</td><td>Intel(R) UHD Graphics 630</td></tr>
<tr><td>GPU Driver</td><td>4.6.0 - Build 26.20.100.7584</td></tr>
<tr><td>Java version</td><td>17.0.5+8</td></tr>
<tr><td>OS arch</td><td>amd64</td></tr>
<tr><td>OS name</td><td>Windows 11</td></tr>
<tr><td>OS version</td><td>10.0</td></tr>
<table>

```
internal.node.NodeImpl.produce_value.class java.lang.AssertionError: Assert failed: No such output, input, or property :html exists for node 72057594037928633 of type editor.placeholder-resource/PlaceholderResourceNode
In production: #{}
beh
    at internal.node.NodeImpl.produce_value(node.clj:294)
    at internal.node$node_value.invokeStatic(node.clj:454)
    at internal.system$node_value.invokeStatic(system.clj:354)
    at dynamo.graph$do_node_value.invokeStatic(graph.clj:830)
    at dynamo.graph$node_value.invokeStatic(graph.clj:833)
    at editor.html_view$resource_handler.invokeStatic(html_view.clj:56)
    at editor.html_view$resource_handler.invoke(html_view.clj:50)
    at clojure.core$partial$fn.invoke(core.clj:2624)
    at util.http_server$setup_server_BANG_$reify.handle(http_server.clj:65)
    at com.sun.net.httpserver.Filter$Chain.doFilter(Filter.java:95)
    at sun.net.httpserver.AuthFilter.doFilter(AuthFilter.java:82)
    at com.sun.net.httpserver.Filter$Chain.doFilter(Filter.java:98)
    at sun.net.httpserver.ServerImpl$Exchange$LinkHandler.handle(ServerImpl.java:851)
    at com.sun.net.httpserver.Filter$Chain.doFilter(Filter.java:95)
    at sun.net.httpserver.ServerImpl$Exchange.run(ServerImpl.java:818)
    at sun.net.httpserver.ServerImpl$DefaultExecutor.execute(ServerImpl.java:201)
    at sun.net.httpserver.ServerImpl$Dispatcher.handle(ServerImpl.java:561)
    at sun.net.httpserver.ServerImpl$Dispatcher.run(ServerImpl.java:527)
    at java.lang.Thread.run(Thread.java:833)
```

--- a/editor/src/clj/editor/resource_update.clj
+++ b/editor/src/clj/editor/resource_update.clj
@@ -276,7 +276,7 @@
   ;; /foo -> /foo.lua
   ;; We should not redirect such nodes, because different resource extensions
   ;; might imply different resource node types
-  (filterv #(= (resource/ext (first %)) (resource/ext (second %))) moved))
+  (filterv #(= (resource/type-ext (first %)) (resource/type-ext (second %))) moved))
 
 (defn resource-change-plan [old-nodes-by-path old-node->old-disk-sha256 changes]
   (let [changes (update changes :moved exclude-extension-changes)

--- a/editor/src/clj/editor/resource_update.clj
+++ b/editor/src/clj/editor/resource_update.clj
@@ -271,8 +271,17 @@
                   (resource-moved-case-plan case move-pairs plan-info))
                 move-cases))))
 
-(defn resource-change-plan [old-nodes-by-path old-node->old-disk-sha256 {:keys [added removed changed moved] :as changes}]
-  (let [basis (g/now)
+(defn- exclude-extension-changes [moved]
+  ;; It's possible to add an extension to file when renaming it, e.g.,
+  ;; /foo -> /foo.lua
+  ;; We should not redirect such nodes, because different resource extensions
+  ;; might imply different resource node types
+  (filterv #(= (resource/ext (first %)) (resource/ext (second %))) moved))
+
+(defn resource-change-plan [old-nodes-by-path old-node->old-disk-sha256 changes]
+  (let [changes (update changes :moved exclude-extension-changes)
+        {:keys [added removed changed moved]} changes
+        basis (g/now)
         move-sources (map first moved)
         move-targets (map second moved)
         move-source-paths (into #{} (map resource/proj-path) move-sources)

